kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,395 @@
+    would need to delete a whole partition to overwrite it.
+
+    \param  pos             The start position to write into the flashrom.
+    \param  src             The source buffer.
+    \param  n               The number of bytes to write.
+
+    \return                 Number of bytes written on success, or -1 on
+                            failure.
+
+    \sa syscall_flashrom_read(), syscall_flashrom_delete()
+*/
+int syscall_flashrom_write(uint32_t pos, const void *src, size_t n);
+
+/** \brief   Delete a partition of the flashrom.
+    \ingroup system_calls
+
+    This function returns a flashrom partition to all 1's, so that it may 
+    be rewritten.
+
+    \warning
+    ALL data in the entire partition will be lost.
+
+    \param  pos             The offset from the start of the flashrom you 
+                            want to delete.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+
+    \sa syscall_flashrom_read(), syscall_flashrom_write()
+*/
+int syscall_flashrom_delete(uint32_t pos);
+
+/** \brief   Initialize the GDROM drive.
+    \ingroup system_calls
+
+    This function initializes the GDROM drive. Should be called before any 
+    commands are sent.
+*/
+void syscall_gdrom_init(void);
+
+/** \brief   Reset the GDROM drive.
+    \ingroup system_calls
+
+    This function resets the GDROM drive.
+*/
+void syscall_gdrom_reset(void);
+
+/** \brief   Checks the GDROM drive status.
+    \ingroup system_calls
+
+    This function retrieves the general condition of the GDROM drive. It 
+    populates a provided array with two elements. The first element 
+    indicates the current drive status, and the second element identifies 
+    the type of disk inserted (if any).
+
+    \param  status          The pointer to two 32-bit integers to 
+                            receive extended status information.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_check_drive(void *status);
+
+/** \brief   Send a command to the GDROM command queue.
+    \ingroup system_calls
+
+    This function sends a command to the GDROM queue.
+
+    \note
+    Call syscall_gdrom_run_commands() to run queued commands.
+
+    \param  cmd             The command code (see CMD_* in cdrom.h).
+    \param  params          The pointer to parameter block for the command, 
+                            can be NULL if the command does not take 
+                            parameters.
+
+    \return                 The request id (>=0) on success, or <0 error
+                            code on failure.
+
+    \sa syscall_gdrom_check_command(), syscall_gdrom_run_commands()
+*/
+int syscall_gdrom_send_command(uint8_t cmd, void *params);
+
+/** \brief   Check status of queued command for the GDROM.
+    \ingroup system_calls
+
+    This function checks if a queued command has completed.
+
+    \param  id              The request id (>=0).
```

Comment:
0 is first in queue, N is last in queue? Other way around? I would be explicit.

Reply:
0 is an error from command request. Valid request ID is >= 1.

Reply:
Im not sure. I will research this tonight/tomorrow.

Reply:
@gyrovorbis Its like @DC-SWAT said.  Looking at the bios, its a counter that goes up by 1 every command you send and circles back to 1 on overflow.